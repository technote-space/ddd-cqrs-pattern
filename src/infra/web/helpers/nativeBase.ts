import { useColorMode, useColorModeValue } from 'native-base';

export { useColorMode, useColorModeValue };
