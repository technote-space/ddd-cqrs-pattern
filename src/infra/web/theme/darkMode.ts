import type { Dispatch, Reducer } from '$/web/shared/store';
import type { IDarkModeContext, IDarkMode, StoreContext } from '$/web/theme/darkMode';
import { useSelector } from 'react-redux';
import { singleton } from 'tsyringe';

@singleton()
export class DarkModeContext implements IDarkModeContext {
  private readonly TOGGLE_DARK_MODE = 'TOGGLE_DARK_MODE';

  public getKey(): string {
    return 'darkMode';
  }

  public getInitialState(): StoreContext {
    return { darkMode: true };
  }

  public getReducerMapObject(): Record<string, Reducer<StoreContext>> {
    return {
      [this.TOGGLE_DARK_MODE]: (store) => ({
        ...store,
        darkMode: !store?.darkMode,
      }),
    };
  }

  public persistTargets(): Array<keyof StoreContext> {
    return ['darkMode'];
  }

  public useDarkMode(): boolean {
    return useSelector((state: { setting: StoreContext }) => state.setting.darkMode);
  }

  public toggleDarkMode(dispatch: Dispatch) {
    dispatch({ type: this.TOGGLE_DARK_MODE });
  }
}

@singleton()
export class DarkMode implements IDarkMode {
  // eslint-disable-next-line
  public useColorModeValue<T>(lightModeValue: T, darkModeValue: T): T {
    return lightModeValue;
  }

  public toggleDarkMode(): () => void {
    return () => {
      //
    };
  }
}
