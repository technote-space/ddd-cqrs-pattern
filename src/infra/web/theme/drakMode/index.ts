import type { useDarkMode as useDarkModeType, useToggleDarkMode as useToggleDarkModeType } from '$/web/theme/darkMode';
import { container } from 'tsyringe';

/* istanbul ignore next */
export const useDarkMode: useDarkModeType = () => container.resolve<useDarkModeType>('useDarkMode')();
/* istanbul ignore next */
export const useToggleDarkMode: useToggleDarkModeType = () => container.resolve<useToggleDarkModeType>('useToggleDarkMode')();
