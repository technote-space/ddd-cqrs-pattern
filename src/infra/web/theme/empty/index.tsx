import type { ITheme, Props } from '$/web/theme';
import type { VFC } from 'react';
import { memo } from 'react';
import { singleton } from 'tsyringe';
import { BaseComponent } from '@/web/shared/component';

@singleton()
export class EmptyTheme extends BaseComponent<Props> implements ITheme {
  public constructor() {
    super();
  }

  protected getComponent(): VFC<Props> {
    const component = memo(({ children }) => <div className="empty-theme">
      {children}
    </div>);
    component.displayName = 'EmptyTheme';

    return component;
  }
}
