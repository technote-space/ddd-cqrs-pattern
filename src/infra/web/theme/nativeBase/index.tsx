import type { IContextProvider } from '$/web/shared/contextProvider';
import type { VFC } from 'react';
import type { PropsWithChildren } from 'react';
import { NativeBaseProvider, extendTheme, Box } from 'native-base';
import { memo } from 'react';
import { singleton } from 'tsyringe';

@singleton()
export class NativeBaseThemeProvider implements IContextProvider {
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  private readonly __provider: VFC<PropsWithChildren<any>>;

  public constructor() {
    const theme = extendTheme({
      colors: {
        // https://smart-swatch.netlify.app/
        // primary: {
        //   50: '#E3F2F9',
        //   100: '#C5E4F3',
        //   200: '#A2D4EC',
        //   300: '#7AC1E4',
        //   400: '#47A9DA',
        //   500: '#0088CC',
        //   600: '#007AB8',
        //   700: '#006BA1',
        //   800: '#005885',
        //   900: '#003F5E',
        // },
      },
      config: {
        useSystemColorMode: true,
      },
    });

    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    const provider = memo(({ children }: PropsWithChildren<any>) => {
      return <NativeBaseProvider theme={theme}>
        <Box
          display="flex"
          height="100vh"
          overflow="auto"
          _light={{ bg: '#fffaf0' }}
          _dark={{ bg: '#1a202c' }}
        >
          {children}
        </Box>
      </NativeBaseProvider>;
    });
    provider.displayName = 'NativeBaseThemeProvider';
    this.__provider = provider;
  }

  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  public getProvider(): VFC<PropsWithChildren<any>> {
    return this.__provider;
  }
}
