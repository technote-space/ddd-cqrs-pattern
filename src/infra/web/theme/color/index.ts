import type { useColor as useColorType, ThemeColor } from '$/web/theme/color';
import { container } from 'tsyringe';
import { useDarkMode } from '@/web/theme/drakMode';

export const useColor: useColorType = (key: keyof ThemeColor) => {
  const themeColor = container.resolve<ThemeColor>('ThemeColor');
  const darkMode = useDarkMode();

  return darkMode ? themeColor[key].dark : themeColor[key].light;
};
