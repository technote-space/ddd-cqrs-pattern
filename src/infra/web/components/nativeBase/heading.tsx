import type { Heading as HeadingType } from '$/web/components/heading';
import { Heading as NBHeading } from 'native-base';
import { memo } from 'react';
import { useColor } from '@/web/theme/color';

const Heading: HeadingType = ({ children, style }) => {
  const textColor = useColor('text');
  return <NBHeading style={{ ...{ color: textColor }, ...style }}>
    {children}
  </NBHeading>;
};

Heading.displayName = 'HeadingComponent';
export default memo(Heading);
