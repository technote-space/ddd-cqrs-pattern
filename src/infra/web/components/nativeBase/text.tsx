import type { Text as TextType } from '$/web/components/text';
import { Text as NBText } from 'native-base';
import { memo } from 'react';
import { useColor } from '@/web/theme/color';

const Text: TextType = ({ children, style }) => {
  const textColor = useColor('text');
  return <NBText style={{ ...{ color: textColor }, ...style }}>
    {children}
  </NBText>;
};

Text.displayName = 'TextComponent';
export default memo(Text);
