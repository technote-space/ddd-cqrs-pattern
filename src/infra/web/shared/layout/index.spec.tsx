import renderer from 'react-test-renderer';
import { container } from 'tsyringe';
import { NativeBaseThemeProvider } from '@/web/theme/nativeBase';
import LayoutComponent from '.';

jest.mock('react-redux');

describe('LayoutComponent', () => {
  it('Layoutのコンテキストプロバイダー', () => {
    container.registerInstance('ThemeColor', {
      background: { dark: '#1a202c' },
      text: { dark: '#fff' },
    });
    const mockRender = jest.fn(() => 'LoadingComponent');
    const mockUseLogout = jest.fn();
    const component = new LayoutComponent({ render: mockRender } as never, { useLogout: mockUseLogout } as never);
    const theme = new NativeBaseThemeProvider();
    const Provider = theme.getProvider();

    const tree = renderer.create(<Provider>
      {component.render({ children: 'Children' })}
    </Provider>);
    expect(tree).toMatchSnapshot();
    expect(mockRender).toBeCalledTimes(1);
    expect(mockUseLogout).toBeCalledTimes(1);
  });
});
