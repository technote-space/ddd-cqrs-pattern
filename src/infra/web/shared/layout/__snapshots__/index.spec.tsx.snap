// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LayoutComponent Layoutのコンテキストプロバイダー 1`] = `
<div
  className="css-view-1dbjc4n"
  style={
    Object {
      "WebkitFlex": 1,
      "flex": 1,
      "msFlex": "1 1 0%",
    }
  }
>
  <div
    className="css-view-1dbjc4n"
    style={
      Object {
        "WebkitFlex": 1,
        "flex": 1,
        "msFlex": "1 1 0%",
      }
    }
  >
    <div
      className="css-view-1dbjc4n"
      style={
        Object {
          "backgroundColor": "rgba(255,250,240,1.00)",
          "display": "flex",
          "height": "100vh",
          "overflowX": "auto",
          "overflowY": "auto",
        }
      }
    >
      <main>
        <div
          className="css-view-1dbjc4n"
          style={
            Object {
              "WebkitBoxDirection": "normal",
              "WebkitBoxOrient": "vertical",
              "WebkitFlexDirection": "column",
              "display": "flex",
              "flexDirection": "column",
              "marginLeft": "auto",
              "marginRight": "auto",
              "maxWidth": "800px",
              "msFlexDirection": "column",
            }
          }
        >
          <div
            className="css-view-1dbjc4n"
            style={
              Object {
                "WebkitBoxDirection": "normal",
                "WebkitBoxOrient": "horizontal",
                "WebkitFlexDirection": "row",
                "flexDirection": "row",
                "marginLeft": "auto",
                "msFlexDirection": "row",
              }
            }
          >
            <div
              className="css-cursor-18t94o4 css-view-1dbjc4n"
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              role="button"
              style={
                Object {
                  "WebkitAlignItems": "center",
                  "WebkitBoxAlign": "center",
                  "WebkitBoxDirection": "normal",
                  "WebkitBoxOrient": "horizontal",
                  "WebkitBoxPack": "center",
                  "WebkitFlexDirection": "row",
                  "WebkitJustifyContent": "center",
                  "alignItems": "center",
                  "backgroundColor": "rgba(6,182,212,1.00)",
                  "borderBottomLeftRadius": "4px",
                  "borderBottomRightRadius": "4px",
                  "borderTopLeftRadius": "4px",
                  "borderTopRightRadius": "4px",
                  "cursor": "pointer",
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "msFlexAlign": "center",
                  "msFlexDirection": "row",
                  "msFlexPack": "center",
                  "msTouchAction": "manipulation",
                  "outlineWidth": "0px",
                  "paddingBottom": "8px",
                  "paddingLeft": "12px",
                  "paddingRight": "12px",
                  "paddingTop": "8px",
                  "touchAction": "manipulation",
                }
              }
              tabIndex="0"
            >
              <div
                className="css-view-1dbjc4n"
                style={
                  Object {
                    "WebkitAlignItems": "center",
                    "WebkitBoxAlign": "center",
                    "WebkitBoxDirection": "normal",
                    "WebkitBoxOrient": "horizontal",
                    "WebkitFlexDirection": "row",
                    "alignItems": "center",
                    "flexDirection": "row",
                    "msFlexAlign": "center",
                    "msFlexDirection": "row",
                  }
                }
              >
                <div
                  className="css-view-1dbjc4n"
                >
                  <div
                    className="css-text-901oao"
                    dir="auto"
                    style={
                      Object {
                        "color": "rgba(255,255,255,1.00)",
                        "fontSize": "14px",
                        "fontStyle": "normal",
                        "fontWeight": "500",
                        "letterSpacing": "0px",
                        "lineHeight": "1.5em",
                      }
                    }
                  >
                    ログアウト
                  </div>
                </div>
              </div>
            </div>
            <div
              className="css-view-1dbjc4n"
              style={
                Object {
                  "width": "8px",
                }
              }
            />
            <div
              className="css-cursor-18t94o4 css-view-1dbjc4n"
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              role="button"
              style={
                Object {
                  "WebkitAlignItems": "center",
                  "WebkitAlignSelf": "center",
                  "WebkitBoxAlign": "center",
                  "WebkitBoxPack": "center",
                  "WebkitJustifyContent": "center",
                  "alignItems": "center",
                  "alignSelf": "center",
                  "backgroundColor": "rgba(0,0,0,0.00)",
                  "borderBottomLeftRadius": "4px",
                  "borderBottomRightRadius": "4px",
                  "borderTopLeftRadius": "4px",
                  "borderTopRightRadius": "4px",
                  "cursor": "pointer",
                  "justifyContent": "center",
                  "msFlexAlign": "center",
                  "msFlexItemAlign": "center",
                  "msFlexPack": "center",
                  "msGridRowAlign": "center",
                  "msTouchAction": "manipulation",
                  "outlineWidth": "0px",
                  "paddingBottom": "8px",
                  "paddingLeft": "8px",
                  "paddingRight": "8px",
                  "paddingTop": "8px",
                  "touchAction": "manipulation",
                }
              }
              tabIndex="0"
            >
              <svg
                role="img"
                stroke=""
                style={
                  Object {
                    "color": "rgba(38,38,38,1.00)",
                    "height": "32px",
                    "width": "32px",
                  }
                }
                viewBox="0 0 24 24"
              >
                <g>
                  <path
                    d="M21.4,13.7C20.6,13.9,19.8,14,19,14c-5,0-9-4-9-9c0-0.8,0.1-1.6,0.3-2.4c0.1-0.3,0-0.7-0.3-1 c-0.3-0.3-0.6-0.4-1-0.3C4.3,2.7,1,7.1,1,12c0,6.1,4.9,11,11,11c4.9,0,9.3-3.3,10.6-8.1c0.1-0.3,0-0.7-0.3-1 C22.1,13.7,21.7,13.6,21.4,13.7z"
                    fill="currentColor"
                    stroke=""
                  />
                </g>
              </svg>
            </div>
          </div>
          <div
            className="css-text-901oao"
            dir="auto"
            style={
              Object {
                "color": "rgba(31,41,55,1.00)",
                "fontSize": "14px",
                "fontStyle": "normal",
                "fontWeight": "400",
                "letterSpacing": "0px",
                "lineHeight": "1.5em",
              }
            }
          >
            Children
          </div>
          <div
            className="css-text-901oao"
            dir="auto"
            style={
              Object {
                "color": "rgba(31,41,55,1.00)",
                "fontSize": "14px",
                "fontStyle": "normal",
                "fontWeight": "400",
                "letterSpacing": "0px",
                "lineHeight": "1.5em",
              }
            }
          >
            LoadingComponent
          </div>
        </div>
      </main>
    </div>
  </div>
</div>
`;
